distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -355,3 +504,39 @@ type Date
     == self that =
         sign = Time_Utils.compare_to_localdate self that
         0 == sign
+
+## PRIVATE
+week_days_between start end =
+    ## We split the interval into 3 periods: the first week (containing the
+       starting point), the last week (containing the end point), and the full
+       weeks in between those. In some cases there may be no weeks in-between
+       and the first and last week can be the same week.
+    start_of_first_full_week = Time_Utils.start_of_next_week start
```

Comment:
I feel this calls for `start.start_of Date_Period.Week`

Reply:
Well yes, but we did not implement the `Week` variant as you told me we want to postpone it because there are the complications with the "which day is the start of the week".

I avoided adding it to avoid adding additional features as I feel I sometimes tend to do that and then get the PR delayed because of it. So I decided to go for the simple solution - create a _private_ helper which does only the job we need it to - being much simpler then the whole logic needed for `start_of Date_Period.Week` machinery.

Still - if you think this _is_ the right time to implement `start_of Date_Period.Week` - I'm happy to do that. Do we want to add a `start_of_week=Day_Of_Week.Monday` setting which can be overridden?

Reply:
Still I'd suggest doing it as a separate PR, just to move this forward and avoid mixing features in a single PR. Then we can replace these helper functions here in that next PR.